### PR TITLE
LibJS+LibUnicode: Implement the Extend TimeZoneName Option Proposal

### PR DESCRIPTION
**Pull request by trflynn89**

### CODE REVIEW

**Comment by trflynn89** [comment]
Rebased for the macOS fix

File: Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp [hunk 1]
```diff
@@ -614,7 +617,61 @@ Optional<Unicode::CalendarPattern> basic_format_matcher(Unicode::CalendarPattern
             else if (options_prop.has_value() && !format_prop.has_value()) {
                 score -= removal_penalty;
             }
-            // v. Else if optionsProp ≠ formatProp, then
+            // v. Else if property is "timeZoneName", then
+            else if (type == Unicode::CalendarPattern::Field::TimeZoneName) {
+                // The following will always be true, but is needed to compile.
+                if constexpr (IsSame<ValueType, Unicode::CalendarPatternStyle>) {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp:623]: can we somehow solve this without introducing an unnecessary level of indentation?

**Review by linusg — COMMENTED** [review]
LGTM, one question

File: Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp:623]: We could do something like use a helper lambda for the comparisons:
```c++
else if (type == Unicode::CalendarPattern::Field::TimeZoneName) {
    auto compare_prop = [](auto prop, auto test) {
        if (!prop.has_value())
            return false;
        return static_cast<Unicode::CalendarPatternStyle>(*prop) == test;
    };

    // 1. If optionsProp is "short" or "shortGeneric", then
    if (compare_prop(options_prop, Unicode::CalendarPatternStyle::Short) || compare_prop(options_prop, Unicode::CalendarPatternStyle::ShortGeneric)) {                  
```

**Review by linusg — APPROVED** [review]

**Pull request merged**